plugin/completion.py

```diff
@@ -260,7 +256,14 @@ def do_request(self, prefix: str, locations: 'List[int]') -> None:
         if not client:
             return
 
-        if settings.complete_all_chars or self.is_after_trigger_character(locations[0]):
+        prev_point = locations[0] - 1 if locations[0] - 1 >= 0 else 0
+        prev_char = view.substr(prev_point)
+
+        if (
+            settings.complete_all_chars or
+            prev_char in self.trigger_chars or
+            prev_char not in self.ignored_trigger_chars
+        ):
```

Comment:
This looks strange:

a) If settings.complete_all_chars is true, then the expression in the if-statement is always true, right?
b) If settings.complete_all_chars is false, then the expression in the if-statement should only be true when prev_char in self.trigger_chars is true, but it is also true when prev_char not in self.ignored_trigger_chars, which is most of the time.

I don't think we even need self.ignored_trigger_chars:

```suggestion
    if prev_char in self.trigger_chars or (settings.complete_all_chars and prev_char not in self.view.settings().get("word_separators")):
```

This says: if the previous character is a trigger character, then do the completion dance, otherwise, if complete_all_chars is true and the previous character is not a word separator, do the completion dance.